File: web-ui/src/types/manager/models/PipelineStatus.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

/**
 * Lifecycle of a pipeline.
 */
export enum PipelineStatus {
  SHUTDOWN = 'Shutdown',
  DEPLOYED = 'Deployed',
  RUNNING = 'Running',
  PAUSED = 'Paused',
  FAILED = 'Failed'
}

File: python/dbsp-api-client/dbsp_api_client/models/pipeline_status.py
from enum import Enum


class PipelineStatus(str, Enum):
    FAILED = "Failed"
    INITIALIZING = "Initializing"
    PAUSED = "Paused"
    PROVISIONING = "Provisioning"
    RUNNING = "Running"
    SHUTDOWN = "Shutdown"
    SHUTTINGDOWN = "ShuttingDown"

    def __str__(self) -> str:
        return str(self.value)
